lib/gringotts/gateways/securion_pay.ex

```diff
@@ -0,0 +1,217 @@
+      iex> result = Gringotts.Gateways.SecurionPay.authorize(amount, card, opts)
+
+  ## Example
+  ### With a `card_token` and `customer_token`
+      iex> amount = Money.new(20, :USD}
+      iex> opts = [config: [:secret_key: "c2tfdGVzdF9GZjJKcHE1OXNTV1Q3cW1JOWF0aWk1elI6"], customer_id: "cust_zpYEBK396q3rvIBZYc3PIDwT"]
+      iex> card = "card_LqTT5tC10BQzDbwWJhFWXDoP"
+      iex> result = Gringotts.Gateways.SecurionPay.authorize(amount, card, opts)
+
+  """
+  @spec authorize(Money.t(), CreditCard.t() | String.t(), keyword) :: {:ok | :error, Response}
+
+  def authorize(amount, %CreditCard{} = card, opts) do
+    header = [{"Authorization", "Basic " <> opts[:config][:secret_key]}]
+    token_id = create_token(card, header)
+    {currency, value, _, _} = Money.to_integer_exp(amount)
+
+    token_id
+    |> create_params(currency, value, false)
+    |> commit(:post, "charges", header)
+    |> respond
+  end
+
+  def authorize(amount, card_id, opts) when is_binary(card_id) do
+    header = [{"Authorization", "Basic " <> opts[:config][:secret_key]}]
+    {currency, value, _, _} = Money.to_integer_exp(amount)
+    params = create_params(card_id, opts[:customer_id], currency, value, false)
+
+    params
+    |> commit(:post, "charges", header)
+    |> respond
+  end
+
+  ###############################################################################
+  #                                PRIVATE METHODS                              #
+  ###############################################################################
+
+  # Creates the parameters for authorise function when 
+  # card_id and customerId is provided.
+  @spec create_params(String.t(), String.t(), String.t(), Integer.t(), boolean) :: {[]}
+  defp create_params(card_id, customer_id, currency, value, captured) do
+    [
+      {"amount", value},
+      {"currency", to_string(currency)},
+      {"card", card_id},
+      {"captured", "#{captured}"},
+      {"customerId", customer_id}
+    ]
+  end
+
+  # Creates the parameters for authorise when token is provided.
+  @spec create_params(String.t(), String.t(), Integer.t(), boolean) :: {[]}
+  defp create_params(token, currency, value, captured) do
+    [
+      {"amount", value},
+      {"currency", to_string(currency)},
+      {"card", token},
+      {"captured", "#{captured}"}
+    ]
+  end
+
+  # Makes the request to SecurionPay's network.
+  # For consistency with other gateway implementations, make your (final)
+  # network request in here, and parse it using another private method called
+  # `respond`.
+  defp commit(params, method, path, header) do
+    HTTPoison.request(method, "#{@base_url}#{path}", {:form, params}, header)
+  end
+
+  # Parses SecurionPay's response and returns a `Gringotts.Response` struct
+  # in a `:ok`, `:error` tuple.
+  @spec respond(term) :: {:ok | :error, Response}
+
+  defp respond({:ok, %{status_code: 200, body: body}}) do
+    parsed_body = Poison.decode!(body)
+
+    {:ok,
+     %{
+       success: true,
```

Comment:
Respond function should return response of predefined format as in Gringotts.Response.ex